shell/exec.go

```diff
@@ -102,8 +103,24 @@ func (s *MenderShell) pipeStdout() {
 		raw := make([]byte, 255)
 		n, err := sr.Read(raw)
 		if err != nil {
-			log.Errorf("error reading stdout: '%s'; restart is needed.", err)
-			break
+			log.Errorf("error reading stdout: %s", err)
+			//
+			msg := &ws.ProtoMsg{
+				Header: ws.ProtoHdr{
+					Proto:     ws.ProtoTypeShell,
+					MsgType:   wsshell.MessageTypeStopShell,
+					SessionID: s.sessionId,
+					Properties: map[string]interface{}{
+						"status": wsshell.NormalMessage,
```

Comment:
I'm not entirely sure about the `wsshell` specifics, but I believe it should be:
```suggestion
						"status": wsshell.ErrorMessage,
```

Reply:
Technically, it is not an error but a "StopShell" command from the device.
We use ErrorMessage in response to a message, don't we?

Reply:
TBH, I don't quite understand the semantics behind the "status" field. Is it required for all shell messages? @merlin-northern ?

Reply:
we are using status field as a status indicator, i.e.: if there is an error, we set "status" to indicate the error for the other side.

Reply:
it is not an error message, so status NormalMessage makes sense to me.
suggestion: can we make a helper function out of it, and provide unit tests for it? let's find a nice place and a nice name `SendMessage(status,body)`
just a suggestion.